hedera-node/src/main/java/com/hedera/services/txns/crypto/CryptoAdjustAllowanceTransitionLogic.java

```diff
@@ -74,33 +78,31 @@ public CryptoAdjustAllowanceTransitionLogic(
 		this.adjustAllowanceChecks = allowanceChecks;
 		this.dynamicProperties = dynamicProperties;
 		this.sideEffectsTracker = sideEffectsTracker;
+		this.entitiesChanged = new HashMap<>();
 	}
 
 	@Override
 	public void doStateTransition() {
 		/* --- Extract gRPC --- */
 		final TransactionBody cryptoAdjustAllowanceTxn = txnCtx.accessor().getTxn();
-		final AccountID owner = cryptoAdjustAllowanceTxn.getTransactionID().getAccountID();
+		final AccountID payer = cryptoAdjustAllowanceTxn.getTransactionID().getAccountID();
 		final var op = cryptoAdjustAllowanceTxn.getCryptoAdjustAllowance();
+		entitiesChanged.clear();
 
 		/* --- Use models --- */
-		final Id ownerId = Id.fromGrpcAccount(owner);
-		final var ownerAccount = accountStore.loadAccount(ownerId);
+		final Id payerId = Id.fromGrpcAccount(payer);
+		final var payerAccount = accountStore.loadAccount(payerId);
 
 		/* --- Do the business logic --- */
-		adjustCryptoAllowances(op.getCryptoAllowancesList(), ownerAccount);
-		adjustFungibleTokenAllowances(op.getTokenAllowancesList(), ownerAccount);
-		adjustNftAllowances(op.getNftAllowancesList(), ownerAccount);
-
-		/* --- validate --- */
-		if (exceedsAccountLimit(ownerAccount)) {
-			sideEffectsTracker.reset();
-			throw new InvalidTransactionException(MAX_ALLOWANCES_EXCEEDED);
-		}
+		adjustCryptoAllowances(op.getCryptoAllowancesList(), payerAccount);
+		adjustFungibleTokenAllowances(op.getTokenAllowancesList(), payerAccount);
+		adjustNftAllowances(op.getNftAllowancesList(), payerAccount);
 
 		/* --- Persist the owner account --- */
-		accountStore.commitAccount(ownerAccount);
-		sigImpactHistorian.markEntityChanged(ownerId.num());
+		for (final var entry : entitiesChanged.entrySet()) {
+			accountStore.commitAccount(entry.getValue());
+			sigImpactHistorian.markEntityChanged(entry.getKey());
```

Comment:
We only need to call `sigImpactHistorian.markEntityChanged()` if we are affecting the **signing requirements of a future transaction**.
 
Afaict, adjusting an account's allowances will not change the signing requirements of any future transaction; so I think we can delete this line.

Reply:
Ahh, I see. this line is a carry forward from master. Thanks , will remove it